libzim/libzim.pyx

```diff
@@ -0,0 +1,286 @@
+cimport libzim
+cimport cpython.ref as cpy_ref
+from cython.operator import dereference
+
+from libc.stdint cimport uint32_t, uint64_t
+from libcpp.string cimport string
+from libcpp cimport bool
+from libcpp.memory cimport shared_ptr, make_shared
+
+import datetime
+
+from collections import defaultdict
+
+#########################
+#       ZimBlob         #
+#########################
+
+cdef class ZimBlob:
+    cdef Blob* c_blob
+
+    def __init__(self, content):
+
+        if isinstance(content, str):
+            ref_content = content.encode('UTF-8')
+        else:
+            ref_content = content
```

Comment:
We should not assume anything about the content.
It is bytes and we don't want to try to encode a string.
(Explicit is better than implicit)

Reply:
This allows to construct the blob either from bytes or str. So that when implementing articles you can do:

```python3
 def get_data(self):
        return ZimBlob("<html> hola <\html>")
```
and will work, otherwise you will need to do for normal content strings as most html is.

 def get_data(self):
        return ZimBlob("<html> hola <\html>".encode('UTF-8'))

Reply:
Hum, yes.
I tend to have all the API "exchanging data" to use bytes. Because this is how things are exchanged and you cannot assume (most of the time) the encoding.
However, the zim format specify that the encoding must be utf8, so we are somehow on a specific case here.
Let's move on with this, we keep the implicit encoding of a str.

Reply:
Well this implicit dual case forces the user to also handle the two cases. Could we keep this api call bytes only and maybe add a `text_content` that automatically decodes UTF-8?

Reply:
We are on the writer part here (the `get_data` is called by the writer to know the data to write in the zim file).

But yes, on the reader part, we shouldn't have a method that some time return bytes and sometime str.
The "base" method should return only bytes (without trying to decode anything) and we could add a "convenient" method to get a decoded str.
But I'm not sure we need to have a second method. It will be explicit in any case (call the base method or the convenient one) and if it is explicit, it is better to let the user code call `decode()` when it needs it.
Anyway, this is a discussion to have one the reader part, not here :)